gramps/plugins/gramplet/agestats.py

```diff
@@ -30,273 +31,264 @@
-        width = self.chart_width
-        graph_width = width - 8
-        self.create_bargraph(age_dict, age_handles, _("Lifespan Age Distribution"), _("Age"), graph_width, 5, self.max_age)
-        self.create_bargraph(father_dict, father_handles, _("Father - Child Age Diff Distribution"), _("Diff"), graph_width, 5, self.max_father_diff)
-        self.create_bargraph(mother_dict, mother_handles, _("Mother - Child Age Diff Distribution"), _("Diff"), graph_width, 5, self.max_mother_diff)
-        start, end = self.gui.buffer.get_bounds()
-        self.gui.buffer.apply_tag_by_name("fixed", start, end)
-        self.append_text("", scroll_to="begin")
 
-    def ticks(self, width, start=0, stop=100, fill=" "):
-        """ Returns the tickmark numbers for a graph axis """
-        count = int(width / 10.0)
-        retval = "%-3d" % start
-        space = int((width - count * 3) / float(count - 1))
-        incr = (stop - start) / float(count - 1)
-        lastincr = 0
-        for i in range(count - 2):
-            retval += " " * space
-            newincr = int(start + (i + 1) * incr)
-            if newincr != lastincr:
-                retval += "%3d" % newincr
-            else:
-                retval += " | "
-            lastincr = newincr
-        rest = width - len(retval) - 3 + 1
-        retval += " " * rest
-        retval += "%3d" % int(stop)
-        return retval
+        self.create_histogram(age_dict, age_handles,
+                              _("Lifespan Age Distribution"),
+                              _("Age"), 5, self.max_age)
+        self.create_histogram(father_dict, father_handles,
+                              _("Father - Child Age Diff Distribution"),
+                              _("Diff"), 5, self.max_father_diff)
+        self.create_histogram(mother_dict, mother_handles,
+                              _("Mother - Child Age Diff Distribution"),
+                              _("Diff"), 5, self.max_mother_diff)
 
-    def format(self, text, width, align="left", borders="||", fill=" "):
-        """ Returns a formatted string for nice, fixed-font display """
-        if align == "center":
-            text = text.center(width, fill)
-        elif align == "left":
-            text = (text + (fill * width))[:width]
-        elif align == "right":
-            text = ((fill * width) + text)[-width:]
-        if borders[0] is not None:
-            text = borders[0] + text
-        if borders[1] is not None:
-            text = text + borders[1]
-        return text
+    def get_date(self, event_type, person):
+        """
+        Find the birth or death date of a given person.
+        """
+        if event_type == 'BIRTH':
+            ref = person.get_birth_ref()
+        else:
+            ref = person.get_death_ref()
+        if ref:
+            event = self.dbstate.db.get_event_from_handle(ref.ref)
+            date = event.get_date_object()
+            if date.get_year() != 0:
+                return date
```

Comment:
@Nick-Hall 

For dates using different calendars, maybe need
to force to one calendar before calculation?
```diff
return date.to_calendar('gregorian') # or .to_calendar('0') 
# or config.get('preferences.calendar-format-report')
```